File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--programmatically-insert-text-into-word-documents_13.vb
    Friend Sub SelectionInsertText()
        Dim currentSelection As Word.Selection = Application.Selection

        ' Store the user's current Overtype selection
        Dim userOvertype As Boolean = Application.Options.Overtype

        ' Make sure Overtype is turned off.
        If Application.Options.Overtype Then
            Application.Options.Overtype = False
        End If

        With currentSelection

            ' Test to see if selection is an insertion point.
            If .Type = Word.WdSelectionType.wdSelectionIP Then
                .TypeText("Inserting at insertion point. ")
                .TypeParagraph()

            ElseIf .Type = Word.WdSelectionType.wdSelectionNormal Then
                ' Move to start of selection.
                If Application.Options.ReplaceSelection Then
                    .Collapse(Direction:=Word.WdCollapseDirection.wdCollapseStart)
                End If
                .TypeText("Inserting before a text block. ")
                .TypeParagraph()

            Else
                ' Do nothing.
            End If
        End With

        ' Restore the user's Overtype selection
        Application.Options.Overtype = userOvertype
    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--programmatically-insert-text-into-word-documents_13.cs
        private void SelectionInsertText() 
        { 
            Word.Selection currentSelection = Application.Selection; 

            // Store the user's current Overtype selection
            bool userOvertype = Application.Options.Overtype;

            // Make sure Overtype is turned off.
            if (Application.Options.Overtype) 
            { 
                Application.Options.Overtype = false; 
            } 

            // Test to see if selection is an insertion point.
            if (currentSelection.Type == Word.WdSelectionType.wdSelectionIP) 
            { 
                currentSelection.TypeText("Inserting at insertion point. ");
                currentSelection.TypeParagraph(); 
            } 
            else 
                if (currentSelection.Type == Word.WdSelectionType.wdSelectionNormal)
                { 
                    // Move to start of selection.
                    if (Application.Options.ReplaceSelection)
                    { 
                        object direction = Word.WdCollapseDirection.wdCollapseStart;
                        currentSelection.Collapse(ref direction);
                    }
                    currentSelection.TypeText("Inserting before a text block. ");
                    currentSelection.TypeParagraph();
                }
                else
                {
                    // Do nothing.
                }

            // Restore the user's Overtype selection
            Application.Options.Overtype = userOvertype;
        }